### DISCUSSION TITLE
Background task function does not execute!

### DISCUSSION THREAD

**Discussion by wjs2063** [fastapi/fastapi#11235] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python

from fastapi import FastAPI, Depends, WebSocket, BackgroundTasks
import async_timeout
import redis.asyncio as redis
import asyncio
import json
import aiohttp

STOPWORD = "STOP"
app = FastAPI()


async def call_external_server(channel, text):
    print("call_external_server start")
    async with aiohttp.ClientSession() as session:
        async with session.get(f"http://127.0.0.1:9000/pub?channel={channel}&text={text}") as resp:
            print(resp) 
    print("call_external_server finished")
    return {"response": "external_server is done"}


@app.websocket("/ws")
async def websocket_endpoint(channel: str, websocket: WebSocket, background_task: BackgroundTasks):
    await websocket.accept()
    client_info = dict(websocket.headers)
    text = client_info.get("text")
    print(text)
    redis_reader: redis.client.PubSub = await get_redis_pubsub()
    await redis_reader.subscribe(channel)
    # Add Background task 
    background_task.add_task(call_external_server, channel, text)
    
    try:
        while True:
            message = await redis_reader.get_message(ignore_subscribe_messages=True)
            if message is not None:
                decoded_msg = message["data"].decode()
                if decoded_msg == STOPWORD:
                    print("(Reader) STOP")
                    break
                print(decoded_msg)
                await websocket.send_text(decoded_msg)
    except Exception as e:
        print(e)
        await websocket.close()
        return
    await websocket.close()
    return
```

### Description

I want to execute the next code immediately without waiting for Backgroundtask function(call_external_server_ response.
then immediately waiting for subscribe redis-channel 

But Background task function doesn't execute 

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.3

### Python Version

Python 3.12.0

### Additional Context

Apple m2 sonoma 14.2.1

**Comment by wjs2063** [comment]
It seems to background tasks execution time is when endpoint handler function is done 
so asyncio.create_task(call_external_server(channel,task)) is worked

**Comment by Kludex** [comment]
It doesn't make sense for background tasks to execute after websocket is done.

I think the right behavior here would be to error.

**Reply by wjs2063** [reply, marked as answer]
Yes it's my mistake, 
Backgroundtask excute after endpoint function, so asyncio.create_task can be the answer
